content/kotlin/concepts/loops/loops.md

```diff
@@ -0,0 +1,107 @@
+---
+Title: 'Loops'
+Description: 'A loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Loops'
+  - 'While'
+  - 'For'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+In Kotlin, a loop is a control flow statement that allows you to repeat a block of code a certain number of times or until a certain condition is met. Kotlin provides several types of loops, including:
+
+- `for` loop: This loop iterates over a range of values or elements in a collection.
+- `while` loop: This loop repeatedly executes a block of code as long as a certain condition is true.
+- `do-while` loop: This loop is similar to a while loop, but it guarantees that the block of code will be executed at least once before the condition is checked.
+
+## For loop
+
+The `for` loop is used to iterate over a range of values or an iterable collection. The syntax for a `for` loop in Kotlin is:
+
+```pseudo
+fun main(){
+    for (item in collection) {
+   // code to be executed for each item
+    }
+}
+```
+
+Here's an example of a `for` loop in Kotlin that prints the numbers from 1 to 10:
+
+```kotlin
+fun main(){
+    for(i in 1..10){
+        print(i)
+    }
+}
+```
+
+The output for the above code will be:
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## While loop
+
+The `while` loops used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
+
+```pseudo
+    fun main(){
+        while(condition){
+            //code to be execute while the condition is true
+        }
+    }
+```
+
+Here's an example of a `while` loop in Kotlin that prints the numbers from 1 to 10:
+
+```kotlin
+    fun main(){
+    var i = 1
+    while (i <= 10) {
+    print(" " + i)
+    i++
+    }
+}
+```
+
+The output for the above code will be:
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## Do-While loop
+
+A `do-while` loop in Kotlin is similar to a `while` loop, but the block of code is always executed at least once before the condition is checked. The syntax for a `do-while` loop is:
+
+```pseudo
+fun main(){
+    do {
+    // code to be executed at least once
+    } while (condition)
```

Comment:
Can we do 2-space indents here?

Reply:
@yangc95 I write in intellij see it take default 1 space indent see I will attach ss also down below.
<img width="311" alt="image" src="https://user-images.githubusercontent.com/94860622/213765917-8696f63e-6ef5-41be-919c-be584fd741b9.png">